pkg/middleware/request_metrics.go

```diff
@@ -108,16 +127,36 @@ func RequestMetrics(features featuremgmt.FeatureToggles, cfg *setting.Cfg, promR
 			histogram := httpRequestDurationHistogram.
 				WithLabelValues(labelValues...)
 
+			var nativeHistogram prometheus.Observer
+			if features.IsEnabled(featuremgmt.FlagEnableNativeHTTPHistogram) {
+				nativeHistogram = nativeHTTPRequestDurationHistogram.
+					WithLabelValues(labelValues...)
+			}
+
+			elapsedTime := time.Since(now).Seconds()
+
 			if traceID := tracing.TraceIDFromContext(r.Context(), true); traceID != "" {
 				// Need to type-convert the Observer to an
 				// ExemplarObserver. This will always work for a
 				// HistogramVec.
 				histogram.(prometheus.ExemplarObserver).ObserveWithExemplar(
-					time.Since(now).Seconds(), prometheus.Labels{"traceID": traceID},
+					elapsedTime, prometheus.Labels{"traceID": traceID},
 				)
+
+				if features.IsEnabled(featuremgmt.FlagEnableNativeHTTPHistogram) {
+					nativeHistogram.(prometheus.ExemplarObserver).ObserveWithExemplar(
+						elapsedTime, prometheus.Labels{"traceID": traceID},
+					)
+				}
+
 				return
```

Comment:
there's a bug here - the later `switch` will be skipped if a traceID is present, so we're missing all of the `count*Requests` calls!

Reply:
:) I actually noticed this but decided to ignore it since the other metrics are kinda pointless and should be deprecated.

Reply:
Fixed!